### PR TITLE
feat(get_gene_panel): refactoring

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- see above

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [x] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [x] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: t/set_gene_panel.t [hunk 1]
```diff
@@ -127,7 +120,26 @@ trap {
 like(
     $trap->stderr,
     qr/Unable \s+ to \s+ write \s+ select_file/xms,
-    q{Throw fatal log message}
+    q{Throw warning log message}
 );
 
+## Given a file without gene_panle information
```
Comment (henrikstranneheim) [inline t/set_gene_panel.t:126]:
```suggestion
## Given a file without gene_panel information
```

File: t/set_gene_panel.t [hunk 2]
```diff
@@ -127,7 +120,26 @@ trap {
...
+            ),
+            aggregate_gene_panels_key => $aggregate_gene_panels_key,
+            recipe_name               => $recipe_name_test,
+            sample_info_href          => \%sample_info,
+        }
+    )
+};
+
+## Then throw fatal message and exit
+like( $trap->stderr, qr/FATAL/xms,  q{Throw fatal log message} );
```
Comment (henrikstranneheim) [inline t/set_gene_panel.t:142]: Is there something more specific to match than the generic FATAL?

**Review by henrikstranneheim — APPROVED** [review]
Nice refactor!

File: lib/MIP/Language/Perl.pm [hunk 3]
```diff
@@ -500,6 +501,23 @@ sub _get_fastq_read_length {
...
+sub _get_gene_panel_info {
+
+## Function : Return gene panel data from header
+## Returns  : $gene_panel_info_regexp
+## Arguments:
+
+    my ($arg_href) = @_;
+
+    # If line starts with gene panel comment
+    my $gene_panel_info_regexp = q?'if (/ \A \#\# (gene_panel= .*) \n /xms ){ ?;
```
[the remark this answers is not in the document]
Comment (jemten) [reply lib/MIP/Language/Perl.pm:513]: Nice! I need to start to use character classes more

**Pull request merged**